src/vs/workbench/parts/files/electron-browser/fileActions.ts

```diff
@@ -1389,23 +1389,12 @@ export function validateFileName(parent: IFileStat, name: string, allowOverwriti
 	return null;
 }
 
-function alreadyExists(parent: IFileStat, name: string): { exists: boolean, child: IFileStat | undefined } {
-	let duplicateChild: IFileStat;
+function alreadyExists(parent: ExplorerItem, name: string): { exists: boolean, child: ExplorerItem | undefined } {
+	let duplicateChild: ExplorerItem;
 
 	if (parent.children) {
-		let exists: boolean = parent.children.some((c) => {
-			let found: boolean;
-			if (isLinux) {
-				found = c.name === name;
-			} else {
-				found = c.name.toLowerCase() === name.toLowerCase();
-			}
-			if (found) {
-				duplicateChild = c;
-			}
-			return found;
-		});
-		return { exists, child: duplicateChild };
+		duplicateChild = parent.children[isLinux ? name : name.toLowerCase()];
```

Comment:
@isidorn this could be a utility method that takes care of the platform specific lowercasing (e.g. `stat.hasChild`)